keps/sig-release/3031-signing-release-artifacts/README.md

```diff
@@ -0,0 +1,165 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Target of this enhancement is to define which technology the Kubernetes
+community is using to signs release artifacts.
+
+## Motivation
+
+Signing artifacts provides end users a chance to verify the integrity of the
+downloaded resource. It allows to mitigate man-in-the-middle attacks directly on
+the client side and therefore ensures the trustfulness of the remote serving the
+artifacts.
+
+### Goals
+
+- Defining the used tooling for signing all Kubernetes related artifacts
+- Providing a standard signing process for related projects (like k/release)
+
+### Non-Goals
+
+- Discussing not user-facing internal technical implementation details
+
+## Proposal
+
+Every Kubernetes release produces a set of artifacts. We define artifacts as
+something consumable by end users. Artifacts can be binaries, container images,
+checksum files, documentation, provenance metadata, or the software bill of
+materials. None of those end-user resources are signed right now.
+
+The overall goal of SIG Release is to unify the way how to sign artifacts. This
+will be done by relying on the tools of the Linux Foundations digital signing
+project [sigstore](https://www.sigstore.dev). This goal aligns with the
+[Roadmap and Vision](https://github.com/kubernetes/sig-release/blob/f62149/roadmap.md)
+of SIG Release to provide a secure software supply chain for Kubernetes. It also
+joins the effort of gaining full SLSA Compliance in the Kubernetes Release
+Process ([KEP-3027](https://github.com/kubernetes/enhancements/issues/3027)).
+Because of that, the future [SLSA](https://slsa.dev) compliance of artifacts
+produced by SIG release will require signing artifacts starting from level 2.
+
+[cosign](https://github.com/sigstore/cosign) will be the tool of our choice when
+speaking about the technical aspects of the solution. How we integrate the
+projects into our build process in k/release is out of scope of this KEP and
+will be discussed in the Release Engineering subproject of SIG Release. A
+pre-evaluation of the tool has been done already to ensure that it meets the
+requirements.
+
+An [ongoing discussion](https://github.com/kubernetes/release/issues/2227) about
```

Comment:
Would be a great pioneering use case for [sigstore trust delegations](https://docs.google.com/document/d/1lNW2lNu9QmDhCgXJaNTYxBUzCjuWS2wj8fCHSadbzxQ/edit?usp=sharing)!

Reply:
@trishankatdatadog Unfortunately, I don't have access to this document. Is this anything we have to consider for this KEP?

Reply:
No, I don't think so, I think that's for another KEP that details the specifics of signing and verification. It can be done with cosign in any case, which you have already mentioned here.

Reply:
No, I don't think so, I think the details can go into the specific KEP that details signing/verification. Thanks!